data/shaders/opengl/grid.frag

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2008-2022 Pioneer Developers. See AUTHORS.txt for details
+// Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+// Adapted from https://ourmachinery.com/post/borderland-between-rendering-and-editor-part-1/
+// Credit to Tobias Persson for providing an example implementation.
+
+#include "attributes.glsl"
+
+layout(std140) uniform GridData {
+	vec4 color_lod0;
+	vec4 color_lod1;
+	vec4 grid_info;
+	vec2 grid_size;
+	float line_width_px;
+};
+
+in vec3 v_eyepos;
+in vec2 v_uv0;
+
+out vec4 frag_color;
+
+float log10(float v)
+{
+	return log(v) / log(10.0);
+}
+
+// Calculate the combined XY line coverage values for the given pixel with the defined grid setting
+float calc_grid_coverage(vec2 uv, vec2 line_width, float cell_size)
+{
+	vec2 cover = 1 - abs(clamp(mod(uv + line_width * 0.5, cell_size) / line_width, 0, 1) * 2 - 1);
+	return max(cover.x, cover.y);
+}
+
+void main(void)
+{
+	// Calculate the screen-space derivatives of the UV coordinates
+	vec4 derivatives = vec4(dFdx(v_uv0), dFdy(v_uv0));
+	vec2 dudv = vec2(length(derivatives.xz), length(derivatives.yw));
```

Comment:
are this two lines similar to the expression:
`vec2 dudv = vec2(length(dFdx(v_uv0)), length(dFdy(v_uv0)));`